Trial Dusk and Trial Night Crashes

"Trial Day" works fine.

When moving towards enemies on "Trial Dusk" and "Trial Night", the game crashes.
Using latest Git commits on Main (commit e4dd824). 
Attempted with Timber Wolf and Fire Moth. Probably other mechs will do the same.

Console Output:

``Initializing Camera
panic: runtime error: invalid memory address or nil pointer dereference
[signal SIGSEGV: segmentation violation code=0x1 addr=0x8 pc=0x955840]

goroutine 40 [running]:
github.com/pixelmek-3d/pixelmek-3d/game.(*Game).updatePlayer(0xc000004f00)
        /home/kwadroke/code/pixelmek-3d/game/game.go:491 +0xb40
github.com/pixelmek-3d/pixelmek-3d/game.(*GameScene).Update(0xc002cc8270)
        /home/kwadroke/code/pixelmek-3d/game/scene_game.go:82 +0x111
github.com/pixelmek-3d/pixelmek-3d/game.(*Game).Update(0xc000004f00)
        /home/kwadroke/code/pixelmek-3d/game/game.go:302 +0x3b
github.com/hajimehoshi/ebiten/v2.(*gameForUI).Update(0xc000333e30)
        /home/kwadroke/go/pkg/mod/github.com/hajimehoshi/ebiten/v2@v2.6.6/gameforui.go:142 +0x23
github.com/hajimehoshi/ebiten/v2/internal/ui.(*context).updateFrameImpl(0xc00048b680, {0x24903f0?, 0xc000149200}, 0x1, 0x408e000000000000, 0x4088000000000000, 0x7fcefc743f18?, 0x29bae00, 0x0?, 0xc0006d7560)
        /home/kwadroke/go/pkg/mod/github.com/hajimehoshi/ebiten/v2@v2.6.6/internal/ui/context.go:145 +0x3f2
github.com/hajimehoshi/ebiten/v2/internal/ui.(*context).updateFrame(0xc00071d800?, {0x24903f0, 0xc000149200}, 0xc000097798?, 0xc000358000?, 0xc0000976e0?, 0xc000486eb0?, 0xc000486eb0?)
        /home/kwadroke/go/pkg/mod/github.com/hajimehoshi/ebiten/v2@v2.6.6/internal/ui/context.go:75 +0x89
github.com/hajimehoshi/ebiten/v2/internal/ui.(*userInterfaceImpl).updateGame(0x29bae00)
        /home/kwadroke/go/pkg/mod/github.com/hajimehoshi/ebiten/v2@v2.6.6/internal/ui/ui_glfw.go:1231 +0x1da
github.com/hajimehoshi/ebiten/v2/internal/ui.(*userInterfaceImpl).loopGame(0x29bae00)
        /home/kwadroke/go/pkg/mod/github.com/hajimehoshi/ebiten/v2@v2.6.6/internal/ui/ui_glfw.go:1199 +0xac
github.com/hajimehoshi/ebiten/v2/internal/ui.(*userInterfaceImpl).Run.func2()
        /home/kwadroke/go/pkg/mod/github.com/hajimehoshi/ebiten/v2@v2.6.6/internal/ui/run_glfw_notsinglethread.go:60 +0x37
golang.org/x/sync/errgroup.(*Group).Go.func1()
        /home/kwadroke/go/pkg/mod/golang.org/x/sync@v0.6.0/errgroup/errgroup.go:78 +0x56
created by golang.org/x/sync/errgroup.(*Group).Go in goroutine 1
        /home/kwadroke/go/pkg/mod/golang.org/x/sync@v0.6.0/errgroup/errgroup.go:75 +0x96
`

--- a/game/game.go
+++ b/game/game.go
@@ -488,7 +488,7 @@ func (g *Game) updatePlayer() {
 				nav.SetVisited(true)
 
 				// automatically cycle to next nav point
-				if g.player.navPoint.NavPoint == nav && nav.Objective() != model.NavDustoffObjective {
+				if g.player.NavPoint() == nav && nav.Objective() != model.NavDustoffObjective {
 					g.navPointCycle(false)
 				}
 			}
@@ -626,7 +626,7 @@ func (g *Game) navPointCycle(replaceTarget bool) {
 	if replaceTarget && g.player.Target() != nil {
 		// unset player target so status display can show nav selection
 		g.player.SetTarget(nil)
-		if g.player.navPoint != nil {
+		if g.player.currentNav != nil {
 			// on the first time after unset target have it not cycle to next nav
 			return
 		}
@@ -634,7 +634,7 @@ func (g *Game) navPointCycle(replaceTarget bool) {
 
 	var newNav *model.NavPoint
 	navPoints := g.mission.NavPoints
-	currentNav := g.player.navPoint
+	currentNav := g.player.currentNav
 
 	for _, n := range navPoints {
 		if currentNav == nil {
@@ -653,7 +653,7 @@ func (g *Game) navPointCycle(replaceTarget bool) {
 		newNav = navPoints[0]
 	}
 
-	g.player.navPoint = render.NewNavSprite(newNav, 1.0)
+	g.player.currentNav = render.NewNavSprite(newNav, 1.0)
 }
 
 func (g *Game) targetCrosshairs() model.Entity {

--- a/game/hud.go
+++ b/game/hud.go
@@ -374,7 +374,8 @@ func (g *Game) drawTargetStatus(hudOpts *render.DrawHudOptions) {
 
 func (g *Game) drawNavStatus(hudOpts *render.DrawHudOptions) {
 	navStatus := g.GetHUDElement(HUD_NAV_STATUS).(*render.NavStatus)
-	if navStatus == nil || g.player.navPoint == nil || g.player.Target() != nil {
+	navPoint := g.player.NavPoint()
+	if navStatus == nil || navPoint == nil || g.player.Target() != nil {
 		return
 	}
 
@@ -392,7 +393,6 @@ func (g *Game) drawNavStatus(hudOpts *render.DrawHudOptions) {
 		sX, sY, sX+statusWidth, sY+statusHeight,
 	)
 
-	navPoint := g.player.navPoint.NavPoint
 	pPos, nPos := g.player.Pos(), navPoint.Pos()
 	navLine := geom.Line{
 		X1: pPos.X, Y1: pPos.Y,
@@ -469,10 +469,10 @@ func (g *Game) drawCompass(hudOpts *render.DrawHudOptions) {
 		compass.SetTargetHeading(tAngle)
 	}
 
-	if g.player.navPoint == nil {
+	if g.player.currentNav == nil {
 		compass.SetNavEnabled(false)
 	} else {
-		navPos := g.player.navPoint.Pos()
+		navPos := g.player.currentNav.Pos()
 		tLine := geom.Line{
 			X1: playerPos.X, Y1: playerPos.Y,
 			X2: navPos.X, Y2: navPos.Y,
@@ -628,10 +628,7 @@ func (g *Game) drawRadar(hudOpts *render.DrawHudOptions) {
 	playerAngle := g.player.Heading()
 	playerTarget := g.player.Target()
 
-	var playerNav *model.NavPoint
-	if g.player.navPoint != nil {
-		playerNav = g.player.navPoint.NavPoint
-	}
+	playerNav := g.player.NavPoint()
 
 	// discover nav points that are in range
 	navCount := 0
@@ -752,11 +749,11 @@ func (g *Game) drawTargetReticle(hudOpts *render.DrawHudOptions) {
 
 func (g *Game) drawNavReticle(hudOpts *render.DrawHudOptions) {
 	navReticle := g.GetHUDElement(HUD_NAV_RETICLE).(*render.NavReticle)
-	if navReticle == nil || g.player.Target() != nil || g.player.navPoint == nil {
+	if navReticle == nil || g.player.Target() != nil || g.player.currentNav == nil {
 		return
 	}
 
-	s := g.player.navPoint
+	s := g.player.currentNav
 	if s == nil {
 		return
 	}

--- a/game/player.go
+++ b/game/player.go
@@ -41,7 +41,7 @@ type Player struct {
 	selectedWeapon      uint
 	selectedGroup       uint
 	fireMode            model.WeaponFireMode
-	navPoint            *render.NavSprite
+	currentNav          *render.NavSprite
 	ejectionPod         *render.ProjectileSprite
 }
 
@@ -119,6 +119,13 @@ func (p *Player) TurretAngle() float64 {
 		return 0
 	}
 	return p.Unit.TurretAngle()
+}
+
+func (p *Player) NavPoint() *model.NavPoint {
+	if p.currentNav == nil {
+		return nil
+	}
+	return p.currentNav.NavPoint
 }
 
 func (p *Player) CameraPosition() (pos *geom.Vector2, posZ float64) {

--- a/game/sprites.go
+++ b/game/sprites.go
@@ -233,8 +233,8 @@ func (g *Game) getRaycastSprites() []raycaster.Sprite {
 	}
 
 	// add the currently selected nav point as sprite
-	if g.player.navPoint != nil {
-		raycastSprites = append(raycastSprites, g.player.navPoint)
+	if g.player.currentNav != nil {
+		raycastSprites = append(raycastSprites, g.player.currentNav)
 		count++
 	}
 

